### PR TITLE
prov/efa: Fix efa_rdm_readrsp_hdr header size assert

### PR DESCRIPTION
**Pull request by sunkuamzn**

### CODE REVIEW

File: prov/efa/src/rdm/efa_rdm_protocol.h [hunk 1]
```diff
@@ -233,7 +233,7 @@ struct efa_rdm_readrsp_hdr {
 	uint64_t seg_length;
 };
 
-EFA_RDM_ENSURE_HEADER_SIZE(efa_rdm_ctsdata_hdr, sizeof (struct efa_rdm_ctsdata_hdr));
+EFA_RDM_ENSURE_HEADER_SIZE(efa_rdm_readrsp_hdr, 24);
```
Comment (darrylabbate) [inline prov/efa/src/rdm/efa_rdm_protocol.h:236]: Oops. If you want to change it to the previous/intended logic:
```suggestion
EFA_RDM_ENSURE_HEADER_SIZE(efa_rdm_readrsp_hdr, sizeof (struct efa_rdm_ctsdata_hdr));
```

**Review by darrylabbate — APPROVED** [review]
oops

File: prov/efa/src/rdm/efa_rdm_protocol.h (hunk 1 above)
Comment (sunkuamzn) [reply prov/efa/src/rdm/efa_rdm_protocol.h]: fixed

File: prov/efa/src/rdm/efa_rdm_protocol.h [hunk 2]
```diff
@@ -233,7 +233,7 @@ struct efa_rdm_readrsp_hdr {
 	uint64_t seg_length;
 };
 
-EFA_RDM_ENSURE_HEADER_SIZE(efa_rdm_ctsdata_hdr, sizeof (struct efa_rdm_ctsdata_hdr));
+EFA_RDM_ENSURE_HEADER_SIZE(efa_rdm_readrsp_hdr, sizeof(struct efa_rdm_ctsdata_hdr));
```
Comment (shijin-aws) [inline prov/efa/src/rdm/efa_rdm_protocol.h:236]: I don't quite follow... why ensure efa_rdm_readrsp_hdr has a size of efa_rdm_ctsdata_hdr?

Comment (darrylabbate) [reply prov/efa/src/rdm/efa_rdm_protocol.h:236]: This was the oldest commit I could find: https://github.com/ofiwg/libfabric/blame/ab4ef648c1288b0459e4ced9c5363db9dd48b6dd/prov/efa/src/rdm/efa_rdm_protocol.h#L268-L270

No comment on why it needs to be the same size

Comment (shijin-aws) [reply prov/efa/src/rdm/efa_rdm_protocol.h:236]: @wzamazon  is this by design?

Comment (wzamazon) [reply prov/efa/src/rdm/efa_rdm_protocol.h:236]: I think it was because the same function was used to handle READRSP and CTSDATA

Comment (wzamazon) [reply prov/efa/src/rdm/efa_rdm_protocol.h:236]: Now that we have different function to handle them I do not this this assertion is necessary any more.

Comment (sunkuamzn) [reply prov/efa/src/rdm/efa_rdm_protocol.h]: Changed it back to 24 bytes

**Review by shijin-aws — APPROVED** [review]

**Pull request merged**